tests/test_plugin.py

```diff
@@ -261,3 +263,186 @@ def test_crud_for_navigation_plugin_when_versioning_disabled(self):
+        menu_content_version = factories.MenuVersionFactory()
+        menu_content = menu_content_version.content
+        child = factories.ChildMenuItemFactory(parent=menu_content.root)
+        # grandchild
+        factories.ChildMenuItemFactory(parent=child)
+
+        menu_content_version.publish(user=self.get_superuser())
+
+        # Add nav plugin to placeholder
+        self._add_nav_plugin_and_assert(
+            placeholder, menu_content.menu, "menu/menu.html"
+        )
+
+        # Now publish the page content containing the plugin,
+        # so the page can be viewed
+        version = page_content.versions.get()
+        version.publish(self.get_superuser())
+
+        # Making sure there is no cachekey existed before rendering page
+        cache_key = CacheKey.objects.all().count()
+        self.assertEqual(cache_key, 0)
+        # And view the page
+        page_url = page_content.page.get_absolute_url()
+        response = self.client.get(page_url)
+
+        # Rendering should generate cachekey object
+        cache_key = CacheKey.objects.all().count()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(cache_key, 1)
+
+        menu_content_version.unpublish(user=self.get_superuser())
+
+        # Version unpublish action should be invalidated cache_key object
+        cache_key = CacheKey.objects.all().count()
+        self.assertEqual(cache_key, 0)
+
+    def test_menu_cache_invalidate_after_menucontent_archive(self):
+        # NOTE: This test is based on a similar one from django-cms:
+        # https://github.com/divio/django-cms/blob/2daeb7d63cb5fee49575a834d0f23669ce46144e/cms/tests/test_plugins.py#L160
+
+        # Set up a versioned page with one placeholder
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        placeholder = factories.PlaceholderFactory(source=page_content)
+        menu_content_version = factories.MenuVersionFactory()
+        menu_content = menu_content_version.content
+        child = factories.ChildMenuItemFactory(parent=menu_content.root)
+        # grandchild
+        factories.ChildMenuItemFactory(parent=child)
+
+        # Add nav plugin to placeholder
+        self._add_nav_plugin_and_assert(
+            placeholder, menu_content.menu, "menu/menu.html"
+        )
+
+        # Now publish the page content containing the plugin,
+        # so the page can be viewed
+        version = page_content.versions.get()
+        version.publish(self.get_superuser())
+
+        # Making sure there is no cachekey existed before rendering page
+        cache_key = CacheKey.objects.all().count()
+        self.assertEqual(cache_key, 0)
+        # And view the page
+        page_url = page_content.page.get_absolute_url()
+        response = self.client.get(page_url)
+
+        # Rendering should generate cachekey object
```

Comment:
Ditto on strange line spacing/comment not describing the block of code it's attached to.